sos/plugins/lxd.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (C) 2016 Jorge Niedbalski <niedbalski@ubuntu.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, UbuntuPlugin
+
+
+class LXD(Plugin, UbuntuPlugin):
+    """LXD is a containers hypervisor.
+    """
+    plugin_name = 'lxd'
+
+    def setup(self):
+        self.add_copy_spec([
+            "/var/log/lxd",
+            "/var/lib/lxd/lxd.db",
+            "/etc/default/lxd-bridge",
+        ])
+
+        # List of containers available on the machine
+        self.add_cmd_output([
+            "lxc list",
+            "lxc profile list",
+            "lxc image list",
+        ])
+
+        self.add_cmd_output([
+            "find /var/lib/lxd -maxdepth 2 -type d",
+        ], suggest_filename='var-lxd-dirs.txt')
```

Comment:
The recursive ls -lanR for this particular directory iterates over all the rootfs of the LXC containers available on the hosts, making the output really large. I prefer to keep stick to a maxdepth of 2, if this doesn't harms any policy.